mne/io/eyelink/_utils.py

```diff
@@ -5,6 +5,252 @@
 import re
 import numpy as np
 
+from ...utils import _check_pandas_installed
+
+
+def _isfloat(token):
+    """Boolean test for whether string can be of type float.
+
+    Parameters
+    ----------
+    token : str
+        Single element from tokens list.
+    """
+    if isinstance(token, str):
```

Comment:
Instead let's use convenience functions as much as possible, like
```
_validate_type(token, (str, 'numeric'), 'token')
```
Then you can just do your try/except without any type business

Reply:
sure, to clarify, that would be:

```
try:
      _validate_type(token, (str, 'numeric'), 'token')
       return True
except ValueError:
       return False
```

?

Reply:
Currently you raise an error for anything other than `str`, then try to convert to float and return True/False based on floatiness. So I guess it should be:
```
_validate_type(token, str, 'token')
try:
    float(token)
except ValueError:
    return False
else:
    return True
```